app/misc/repo.ts

```diff
@@ -273,61 +351,7 @@ function openRepository() {
         branch = branchParts[branchParts.length - 1];
       })
       .then(function () {
-        //Get the list of branches from the repo
```

Comment:
@sterlinglaird I redesigned the code based on your idea. Please have a look. Thank you.

Reply:
now that I think about it, I feel like I could simply just delete this part since there is already a dedicated function that checks the list.